Labonneguigue/CarND-Capstone-System-Integration
# Traffic light classifier

Note: This is to be reviewed after PR #7 and PR #8 as it is based on the corresponding branches.

This detects and classifies the traffic lights on the camera image.
To test it use:
```bash
roslaunch launch/styx.launch log_level:=debug
```
Check "Camera" and uncheck "Manual". Look at the debug info in the terminal and check that it is consistent.

Note:
Initially I thought about simply fine tuning a classifier on the camera images. But it turned out it was not good enough, I still got around 10-15% of images wrongly classified. I believe this is because it is hard to classify images that vary a lot in size with just a classifier. So finally I use a detector taken from the tensorflow object detection API. It is maintained by the Tensorflow team and part of the standard Tensorflow tools.
